CentralComputing/UDPManager.cpp

```diff
@@ -122,17 +122,22 @@ void UDPManager::connection_monitor( const char * hostname, const char * send_po
   struct pollfd fds[1];
   fds[0].fd = socketfd;
   fds[0].events = POLLIN;
-  int timeout = 1000; // 5 seconds TODO: set timeout
-  
+  uint8_t send_buffer[] = {'A','C','K'};
+  uint8_t buffer2[16];
+
   /*
    * Timeout = -min(p) - min(D1) + T + max(D1) + max(p) 
    * p is processing time
    * D1 is delta 1: the time to receive the python's ping
    * T is the period at which this happens
    */
-
-
-
+  // TODO: Read in the following values from Config file, once that branch is merged into master
+  int T      = 1000;  // milliseconds. Period that Python sends pings 
+  int D1_max = 10;    // milliseconds. Max one way trip time, Python --to-> Pod
+  int D1_min = 2;     // milliseconds. Min one way trip time, Python --to-> Pod
+  int P_max  = 5;     // milliseconds. Max processing time on Pod
+  int P_min  = 1;     // milliseconds. Min processing time on Pod
+  int timeout = T + D1_max + P_max - P_min - D1_min; 
```

Comment:
Need to implement a couple other variables to be able to monitor time. 

First requirement:
Must wait for the first PING before we start to monitor for a disconnection. Can't disconnect if we haven't connected first! 

Most basic case (Only receive PINGs, and send ACKs):
If `poll()` returns `0`, indicating a timeout, then we set a variable to show that we are no longer connected, since we should have received a PING within the timeout.


Advanced case (Receive PINGs and commands):
Say the `poll()` timeout is set to `default_timeout` or something like that.
After waiting `x` amount of time (maybe it was half `default_timeout`), `poll()` receives a command (not a PING) and process it. We still need to wait for the PING, though. We need to dynamically set the `poll()` timeout value to be `default_timeout - x` in order to properly monitor the PING.


We announce *changes* (we don't send constant updates) in the state (our state could be `not_yet_connected`, `connected`, `disconnected`) over the Unified command queue. Some other logic somewhere else takes care of what we need to do with that information. All we need to do is to let them know. 
The possible state changes: `not_yet_connected --goes to--> connected`, `connected --goes to--> disconnected`, and `disconnected --goes to-->connected`.


Once disconnected, it is allowed to reconnect.